### DISCUSSION TITLE
Question regarding distributed study

### DISCUSSION THREAD

**Discussion by ScheiklP** [optuna/optuna#4990] [Q&A] [answered]
Hi,
my setup is the following:
I have a MySQL database running on a machine and a cluster with many nodes that each run an instance of TPE and are connected to that database.

In the documentation, it says:
1. Storage doesn’t store the state of the instance of samplers
2. On each trial, for each parameter, TPE fits one Gaussian Mixture Model (GMM) l(x) to the set of parameter values associated with the best objective values, and another GMM g(x) to the remaining parameter values. It chooses the parameter value x that maximizes the ratio l(x)/g(x).

Which leaves me a bit doubting. Is x considered to be part of the sampler's state?
Does each instance of TPE on the nodes consider all results from the database, or just the results that the individual TPE instance produces?

Cheers,
Paul

**Comment by nzw0301** [comment, marked as answer]
> Does each instance of TPE on the nodes consider all results from the database, or just the results that the individual TPE instance produces?

Yes, I think so. It uses the all result stored on the DB.
